linkerd/cache/src/track.rs

```diff
@@ -1,33 +1,63 @@
 use crate::Handle;
 use linkerd2_stack::{layer, NewService};
-use std::task::{Context, Poll};
+use std::{
+    sync::Arc,
+    task::{Context, Poll},
+};
+use tokio::{sync::Notify, time};
+use tracing::{debug, debug_span, trace};
+use tracing_futures::Instrument;
 
 #[derive(Clone)]
 pub struct NewTrack<N> {
     inner: N,
+    idle: time::Duration,
 }
 
 #[derive(Clone)]
 pub struct Track<T> {
     inner: T,
-    _handle: Handle,
+    notify: Arc<Notify>,
 }
 
-/// === impl NewTrack ===
-
 impl<N> NewTrack<N> {
-    pub fn layer() -> impl layer::Layer<N, Service = NewTrack<N>> + Copy + Clone {
-        layer::mk(|inner| NewTrack { inner })
+    pub fn layer(
+        idle: time::Duration,
+    ) -> impl layer::Layer<N, Service = NewTrack<N>> + Copy + Clone {
+        layer::mk(move |inner| Self { inner, idle })
     }
 }
 
 impl<T, N: NewService<T>> NewService<(Handle, T)> for NewTrack<N> {
     type Service = Track<N::Service>;
 
-    #[inline]
-    fn new_service(&mut self, (_handle, target): (Handle, T)) -> Self::Service {
+    fn new_service(&mut self, (handle, target): (Handle, T)) -> Self::Service {
+        let idle = self.idle;
+        let notify = Arc::new(Notify::new());
+
+        // Create a background task that drops the cache entry's handle when the
+        // service hasn't been notified for an `idle` interval.
+        tokio::spawn({
```

Comment:
I was briefly worried about a potential task leak here when the `Track` service is dropped without notifying the background task...but then I realized that in that case, it'll just time out. So this is fine. Might be worth a comment somewhere to remember this in the future?